instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-wildfly-testing/src/test/groovy/WildflyRestTest.groovy

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+import javax.ws.rs.client.Client
+import javax.ws.rs.client.WebTarget
+import org.glassfish.jersey.client.JerseyClientBuilder
+import org.jboss.arquillian.container.test.api.Deployment
+import org.jboss.arquillian.container.test.api.RunAsClient
+import org.jboss.arquillian.spock.ArquillianSputnik
+import org.jboss.arquillian.test.api.ArquillianResource
+import org.jboss.shrinkwrap.api.ShrinkWrap
+import org.jboss.shrinkwrap.api.asset.EmptyAsset
+import org.jboss.shrinkwrap.api.spec.WebArchive
+import org.junit.runner.RunWith
+import test.CdiRestResource
+import test.EjbRestResource
+import test.RestApplication
+
+@RunWith(ArquillianSputnik)
+@RunAsClient
+class WildflyRestTest extends AgentInstrumentationSpecification {
+
+  @ArquillianResource
+  static URI url
+
+  @Deployment
+  static WebArchive createDeployment() {
+    return ShrinkWrap.create(WebArchive)
+      .addClass(RestApplication)
+      .addClass(CdiRestResource)
+      .addClass(EjbRestResource)
+      .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+  }
+
+  def "test #path"() {
+    when:
+    Client client = JerseyClientBuilder.newClient()
+    WebTarget webTarget = client.target(url)
+
+    String result = webTarget.path(path)
+      .request()
+      .get()
+      .readEntity(String)
+
+    then:
+    result == "hello"
+
+    and:
+    assertTraces(1) {
+      trace(0, 2) {
+        span(0) {
+          name ~/.*\${path}/
```

Comment:
only curious, why don't we know the context path?

Reply:
Because the deployed archive name is an uuid. I guess I could add a jboss-web.xml and try to set context path to something predictable, but doesn't seem like it is worth the effort so I just changed it to extract context path from the uri that arquillian gives us and used that.
As for using smoke tests, the current setup where you need to first push test app and only after it gets merged can push the test because you need the image id from ghcr.io is annoying. The mess with logback-classic and jvmArgs wouldn't be needed if instead of embedded container we would use managed container where arquillian starts the server itself. Then we would need to somehow add proper arguments for agent to arquillian.xml but as test and server are not in the same process then would also need to somehow get spans from server to test which would complicate things.

Reply:
I created #2274, hopefully we can make adding new smoke tests less annoying in the future